action/subchain/datamodel_test.go

```diff
@@ -34,14 +35,27 @@ func TestSubChainState(t *testing.T) {
 }
 
 func TestBlockProofState(t *testing.T) {
-	bp1 := blockProof{
+	bp1 := BlockProof{
+		SubChainAddress: "123",
+		Height:          123,
+		ProducerAddress: "123",
+		Roots: []MerkleRoot{
+			MerkleRoot{
+				Name:  "abc",
+				Value: byteutil.BytesTo32B([]byte("10002")),
```

Comment:
nit: you're converting a non 32B slice into 32B array? Though the root problem is that byteutil.BytesTo32B isn't robust.